File: 0205-isomorphic-strings/solution.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        map<char, char> mpp;
        for(int i=0; i<s.size(); i++){
            if(mpp.find(s[i]) != mpp.end()){
                //means allready present 
                if(mpp[s[i]] != t[i]){
                    return false;
                }
            }
            else{
                mpp[s[i]] = t[i];
            }
        }
        // there may be different key have same valus which is false A/c 
        // q eyes 
        // E.g:- s="badc", t="baba";
        // Output :- false  but A/c our solution so far it will give true
        // or E.g:- s="ab", t="xx";
        // Output:- false but A/c our solution so far it will give true
        // so not handle this case

        set<char> set;
        map<char, char> :: iterator itr;
        for(itr=mpp.begin(); itr!=mpp.end(); itr++){
            set.insert(itr->second);
        }
        if(mpp.size() != set.size()) return false;
        return true;
    }
};

File: 0205-isomorphic-strings/solution.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int  sMap[] = new int[256];
        int tMap[] = new int[256];

        for(int i=0; i<256; i++){
            sMap[i] = -1;
            tMap[i] = -1;
        }

        for(int i=0; i<s.length(); i++){
            if(sMap[s.charAt(i)] == -1 && tMap[t.charAt(i)] == -1){
                sMap[s.charAt(i)] = t.charAt(i);
                tMap[t.charAt(i)] = s.charAt(i);

                System.out.println("sMap[s.charAt(i)] " + sMap[s.charAt(i)] );
                System.out.println("tMap[t.charAt(i)]" + tMap[t.charAt(i)]);
            }else{
                System.out.println("(s.charAt(i) != tMap[t.charAt(i)]  || sMap[s.charAt(i)] !=  t.charAt(i) ) = " + s.charAt(i)  + " " + tMap[t.charAt(i)]  + " - " + sMap[s.charAt(i)] + " " + t.charAt(i) );
                if((s.charAt(i) != tMap[t.charAt(i)]  || sMap[s.charAt(i)] !=  t.charAt(i) )) return false;
            }
        }

        return true;
    }
}
